app/player/player.go

```diff
@@ -2,324 +2,478 @@ package player
+	sdBlob         *stream.SDBlob
+	seenChunks     []ReadableChunk
+	enablePrefetch bool
+}
+
+// ReadableChunk interface describes generic chunk object that Stream can Read() from.
+type ReadableChunk interface {
+	Read(offset, n int, dest []byte) (int, error)
+}
+
+type reflectedChunk struct {
+	body []byte
+}
+
+// GetBlobStore returns default pre-configured blob store.
+func GetBlobStore() *peer.Store {
+	return peer.NewStore(peer.StoreOpts{
+		Address: config.GetRefractorAddress(),
+		Timeout: time.Second * time.Duration(config.GetRefractorTimeout()),
+	})
+}
+
+// NewPlayer initializes an instance with optional BlobStore.
+func NewPlayer(opts *Opts) *Player {
+	if opts == nil {
+		opts = &Opts{}
+	}
+	p := &Player{
+		lbrynetClient: opts.Lbrynet,
+	}
+	if opts.EnableLocalCache {
+		cPath := path.Join(os.TempDir(), "blob_cache")
+		cache, err := InitFSCache(&FSCacheOpts{Path: cPath})
+		if err != nil {
+			Logger.Log().Error("unable to initialize cache: ", err)
+		} else {
+			Logger.Log().Infof("player cache initialized at %v", cPath)
+			p.localCache = cache
+			p.enablePrefetch = opts.EnablePrefetch
+		}
+	}
+	return p
+}
+
+func (p *Player) getLbrynetClient() *ljsonrpc.Client {
+	if p.lbrynetClient != nil {
+		return p.lbrynetClient
+	}
+	sdkRouter := router.New(config.GetLbrynetServers())
+	return ljsonrpc.NewClient(sdkRouter.GetBalancedSDKAddress())
+}
 
-	s, err := newReflectedStream(uri)
+func (p *Player) getBlobStore() store.BlobStore {
+	return GetBlobStore()
+}
+
+// Play delivers requested URI onto the supplied http.ResponseWriter.
+func (p *Player) Play(uri string, w http.ResponseWriter, r *http.Request) error {
+	Logger.streamPlaybackRequested(uri, users.GetIPAddressForRequest(r))
+
+	s, err := p.ResolveStream(uri)
 	if err != nil {
+		Logger.streamResolveFailed(uri, err)
 		return err
 	}
-	err = s.fetchData()
+	Logger.streamResolved(s)
+
+	err = p.RetrieveStream(s)
 	if err != nil {
+		Logger.streamRetrievalFailed(uri, err)
 		return err
 	}
-	s.prepareWriter(w)
+	Logger.streamRetrieved(s)
 
-	Logger.LogF(monitor.F{
-		"stream":    s.URI,
-		"remote_ip": users.GetIPAddressForRequest(req),
-	}).Info("stream requested")
-	ServeContent(w, req, "test", time.Time{}, s)
+	w.Header().Set("Content-Type", s.ContentType)
 
-	return err
-}
+	metrics.PlayerStreamsRunning.Inc()
+	defer metrics.PlayerStreamsRunning.Dec()
+	http.ServeContent(w, r, "stream", s.Timestamp(), s)
 
-func newReflectedStream(uri string) (rs *reflectedStream, err error) {
-	sdkRouter := router.New(config.GetLbrynetServers())
-	client := ljsonrpc.NewClient(sdkRouter.GetBalancedSDKAddress())
-	rs = &reflectedStream{URI: uri}
-	err = rs.resolve(client)
-	return rs, err
+	return nil
 }
 
-func (s reflectedStream) blobNum() int {
-	return len(s.SDBlob.BlobInfos) - 1
+// ResolveStream resolves provided URI by calling the SDK.
+func (p *Player) ResolveStream(uri string) (*Stream, error) {
+	s := &Stream{URI: uri}
+
+	r, err := p.getLbrynetClient().Resolve(uri)
```

Comment:
is a new player created for each handler? We should make sure we are getting a balanced SDK for each call otherwise it could get unbalanced if it continuously reuses the same sdk from the creation of the player object.

Reply:
`getLbrynetClient` just returns a balanced server on every call